### ISSUE TITLE
Confusing options: WebClient.javaScriptEngineEnabled_, WebClient.getOptions().isJavaScriptEnabled()

### ISSUE THREAD

**Issue by cdalexndr** [HtmlUnit/htmlunit#281]
What's the difference between those two, and why is there are two methods of disabling javascript?
```
    public boolean isJavaScriptEnabled() {
        return javaScriptEngineEnabled_ && getOptions().isJavaScriptEnabled();
    }
```
When is javascript engine required and used if javascript is disabled in webclient options?

These two options are confusing and only one should be used.

Note:
Doing some benchmark and I've noticed that javascript engine is initialized even if web client's option disables javascript.
This initialization cost is on the order of parsing my test html, so it's not negligible.

**Comment by cdalexndr** [comment]
After more investigation it seems that the _Javascript_ objects are used to get css styles.
Shouldn't css be decoupled from javascript so that you can safely use css without javascript engine?

Also a performance issue: Javascript engine is initialized every time a page is loaded (a cache should be used). Related TODO:
https://github.com/HtmlUnit/htmlunit/blob/c00619c9b08cb870729b4eb47596f27fdfc727d3/src/main/java/com/gargoylesoftware/htmlunit/TopLevelWindow.java#L77

**Comment by rbri** [comment]
I agree, this is confusing from a first look (like many other areas of the HtmlUnit API :-) ).

There are two different ways to switch off js support
1. create a WebClient with js switched off (javaScriptEngineEnabled_). In this case you get the max performance but there is no way to switch js on later for this WebClient (and parts of the code will not work e.g. #131)

1. You can temporary switch off the js support (e.g. triggering of js events) using the WebClient options. In this case we have to start the js support in the background.

Will be great if you can provide some improvements to the documentation to make it more clear. And Fixes for #131 are also welcome.

**Comment by rbri** [comment]
have done another docu update - will close this.
